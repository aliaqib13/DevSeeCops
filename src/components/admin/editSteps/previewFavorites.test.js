import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PreviewFavorites from './previewFavorites';

describe('previewFavorites', () => {
    let component; const
        props = {
            content: {
                type: 'GreyBox',
                content: {
                    text: [
                        '<p>test test</p>',
                    ],
                    titles: [
                        'grey box',
                    ],
                },
            },
            visible: true,
        };

    beforeEach(() => {
        component = shallow(<PreviewFavorites {...props} />);
    });

    it('Should render successfully', () => {
        const wrapper = component.find('.preview-favorites-modal');
        expect(wrapper.exists()).toBeTruthy();
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Should render component which passed down by props', () => {
        const componentType = props.content.type;
        const savedComponent = component.find(componentType);
        expect(savedComponent.exists()).toBeTruthy();
    });

    it('Should render savedComponent with proper content', () => {
        const componentType = props.content.type;
        const savedComponent = component.find(componentType);
        expect(savedComponent.props().block).toBe(props.content);
    });
});
