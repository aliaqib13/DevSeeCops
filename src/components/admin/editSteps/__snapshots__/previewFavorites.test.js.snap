// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`previewFavorites Should render successfully 1`] = `
<Modal
  className="preview-favorites-modal"
  confirmLoading={false}
  footer={
    Array [
      <Button
        block={false}
        ghost={false}
        htmlType="button"
        loading={false}
        onClick={[Function]}
      >
        Close
      </Button>,
    ]
  }
  maskTransitionName="fade"
  okType="primary"
  onCancel={[Function]}
  title="Preview Component"
  transitionName="zoom"
  visible={true}
  width={1000}
>
  <div
    className="blocks-container"
  >
    <div
      className="box-container"
    >
      <GreyBox
        block={
          Object {
            "content": Object {
              "text": Array [
                "<p>test test</p>",
              ],
              "titles": Array [
                "grey box",
              ],
            },
            "type": "GreyBox",
          }
        }
        preview={true}
      />
    </div>
  </div>
</Modal>
`;
